### PR TITLE
Feat/webpack 5 migration

### PR DESCRIPTION
**Pull request by JulianEstefanHotovo**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,52 +1,90 @@
...
+    path: path.join(__dirname, "dist"),
+    publicPath: "/",
+    filename: `bundle.js`,
+    library: "SwaggerUIKongTheme",
+    libraryTarget: "umd",
   },
-  devServer: {
-    contentBase: './dist'
+  production: {
+    path: path.join(
```
Comment (tkajtoch) [inline webpack.config.js:12]: Please replace it with a standard `dist/` path instead

File: webpack.config.js [hunk 2]
```diff
@@ -1,52 +1,90 @@
...
+      __dirname,
+      "/../kong-portal-templates/workspaces/default/themes/base/assets/js"
+    ),
+    publicPath: "/",
+    filename: `swagger-ui-kong-theme-${hash}.${
```
Comment (tkajtoch) [inline webpack.config.js]: Please make the filename static (without hash or debug)

File: webpack.config.js [hunk 3]
```diff
@@ -1,52 +1,90 @@
...
-    minimize: false
-  } : undefined
-}
+};
+
+module.exports = (env, argv) => {
+  return {
+    entry: "./src/index.js",
+    watch: true,
+    devtool: process.env.DEBUG ? "eval-source-map" : undefined,
```
Comment (tkajtoch) [inline webpack.config.js]: Don't use `process.env` when you're using webpack config function export and you have `env` provided as an argument in line 32

File: webpack.config.js [hunk 4]
```diff
@@ -1,52 +1,90 @@
...
+    plugins: [
+      new webpack.ProvidePlugin({
+        process: "process/browser",
```
Comment (tkajtoch) [inline webpack.config.js]: Please use resolve.fallback instead

File: webpack.config.js [hunk 5]
```diff
@@ -1,52 +1,90 @@
...
+      },
+      alias: {
+        process: "process/browser",
+      },
+    },
+    output: outputs[argv.mode],
+    devServer: {
+      contentBase: "./dist",
+    },
+    optimization: process.env.DEBUG
```
Comment (tkajtoch) [inline webpack.config.js]: Please don't use `process.env`

**Review by pk-hotovo — APPROVED** [review]
LGTM, since i made latest changes and it works :)

**Pull request merged**